src/sage/matrix/matrix_cmr_sparse.pyx

```diff
@@ -443,25 +480,50 @@ cdef class Matrix_cmr_chr_sparse(Matrix_cmr_sparse):
             [ 4  5|42 48 54]
             [-----+--------]
             [ 0  0|-1 -2 -3]
-            sage: M1.two_sum(M2,1,1)
+            sage: M1.two_sum(M2, 1, 1)
             [  1   3| -2  -4  -6]
             [  4   6| -5 -10 -15]
             [-------+-----------]
             [  0   0|  7   8   9]
+            sage: M1.two_sum(M2, 1, 1, nonzero_block="bottom_right")
+            [  1   2   3|  0   0]
+            [-----------+-------]
+            [ 32  40  48|  7   9]
+            [ -8 -10 -12| -1  -3]
         """
         cdef Matrix_cmr_chr_sparse sum, first, second
         cdef CMR_CHRMAT *sum_mat
         first = Matrix_cmr_chr_sparse._from_data(first_mat, immutable=False)
         second = Matrix_cmr_chr_sparse._from_data(second_mat, immutable=False)
-        if column < 0 or column >= first._mat.numColumns:
-            raise ValueError("First marker should be a column index of the first matrix")
-        if row < 0 or row >= second._mat.numRows:
-            raise ValueError("Second marker should be a row index of the second matrix")
-        row_subdivision = []
-        column_subdivision = []
-        row_subdivision.append(first._mat.numRows)
-        column_subdivision.append(first._mat.numColumns - 1)
-        CMR_CALL(CMRtwoSum(cmr, first._mat, second._mat, CMRcolumnToElement(column), CMRrowToElement(row), &sum_mat))
+
+        if nonzero_block == "top_right":
+            column = first_index
+            row = second_index
+            if column < 0 or column >= first._mat.numColumns:
+                raise ValueError("First marker should be a column index of the first matrix")
+            if row < 0 or row >= second._mat.numRows:
+                raise ValueError("Second marker should be a row index of the second matrix")
+            row_subdivision = []
+            column_subdivision = []
+            row_subdivision.append(first._mat.numRows)
+            column_subdivision.append(first._mat.numColumns - 1)
+            first_marker = CMRcolumnToElement(column)
+            second_marker = CMRrowToElement(row)
+        else:
```

Comment:
better to do input checking for this parameter, raising an error if an unknown value is passed